core/src/main/java/jenkins/model/Nodes.java

```diff
@@ -166,17 +168,27 @@ private void persistNode(final @Nonnull Node node)  throws IOException {
      * @param node the node to be updated.
      * @return {@code true}, if the node was updated. {@code false}, if the node was not in the list of nodes.
      * @throws IOException if the node could not be persisted.
+     * @since 1.634
      */
     public boolean updateNode(final @Nonnull Node node) throws IOException {
-        if (node == nodes.get(node.getNodeName())) {
-            Queue.withLock(new Runnable() {
+        boolean exists;
+        try {
+            exists = Queue.withLock(new Callable<Boolean>() {
                 @Override
-                public void run() {
-                    jenkins.trimLabels();
+                public Boolean call() throws Exception {
+                    if (node == nodes.get(node.getNodeName())) {
+                        jenkins.trimLabels();
+                        return true;
+                    }
+                    return false;
                 }
             });
+        } catch (Exception e) {
+            // can never happen
+            exists = false;
+        }
+        if (exists) {
```

Comment:
this is no longer thread safe - as the node may have been removed.
an update to a removed node should fail. (return `false`)